### PR TITLE
Topic/mtlong2/asdplng 50 review profile rsyslog 2

### PR DESCRIPTION
**Pull request by mattlongt**
PR to add updated profile_rsyslog to control repo and base role.

### CODE REVIEW

**Comment by billglick** [comment]
@mattlongt Can you rebase these commits?

File: Puppetfile [hunk 1]
```diff
@@ -21,6 +21,7 @@ mod 'ncsa/profile_motd', tag: 'v0.1.0', git: 'https://github.com/ncsa/puppet-pro
 mod 'ncsa/profile_pam_access', tag: 'v0.0.5', git: 'https://github.com/ncsa/puppet-profile_pam_access'
 mod 'ncsa/profile_puppet_agent', tag: 'v0.1.1', git: 'https://github.com/ncsa/puppet-profile_puppet_agent'
 mod 'ncsa/profile_puppet_master', tag: 'v0.1.1', git: 'https://github.com/ncsa/puppet-profile_puppet_master'
+mod 'ncsa/profile_rsyslog', commit: 'a877ec02fe1b17d6c6e1d68d25eaf1ed9e7a1886', git: 'https://github.com/ncsa/puppet-profile_rsyslog'
```
Comment (billglick) [inline Puppetfile:24]: This seems to have reverted. It should be referencing your recently tagged version instead of a commit.

**Comment by mattlongt** [comment]
@billglick thanks for the note. Just realized I've been rebasing wrong this whole time. Should look better now.

**Review by billglick — CHANGES_REQUESTED** [review]
Not referencing the tag for profile_rsyslog

**Comment by billglick** [comment]
I'm also confused why there appears to not be anything configured in control repo to forward logs to Security. Perhaps I'm overlooking that?

File: site-modules/profile/manifests/base.pp [hunk 2]
```diff
@@ -12,6 +12,7 @@
   include ::profile_motd
   include ::profile_pam_access
   include ::profile_puppet_agent
+  include ::profile::rsyslog::client
```
Comment (billglick) [inline site-modules/profile/manifests/base.pp:15]: Shouldn't this be including `::profile_rsyslog`?

```suggestion
  include ::profile_rsyslog
```

File: Puppetfile (hunk 1 above)
Comment (mattlongt) [reply Puppetfile]: Weird, I'm probably still learning how to rebase the right way and wiped something out. That's been fixed back now.

**Review by billglick — CHANGES_REQUESTED** [review]
I don't think `base.pp` is including the correct class.

File: site-modules/profile/manifests/base.pp (hunk 2 above)
Comment (mattlongt) [reply site-modules/profile/manifests/base.pp:15]: This got wiped out in my rebase again as well.

**Comment by mattlongt** [comment]
@billglick you're not overlooking anything. I just didn't want to make assumptions on where logs are supposed to be directed by default. The relay functionality to security is less useful anyway unless the clients are also configured. For now this is done per site/cluster/node. I've written up some documentation in the wiki on how hiera should be setup to send logs to the right places.

It would be trivial to add forwarding to security to the profile itself, but it might be helpful to wait on that until we can also safely set a variable default on the client portion to send logs to a central node in a site (like an xcat server), which then gets passed on to security.

**Comment by billglick** [comment]
> @billglick you're not overlooking anything. I just didn't want to make assumptions on where logs are supposed to be directed by default. The relay functionality to security is less useful anyway unless the clients are also configured. For now this is done per site/cluster/node. I've written up some documentation in the wiki on how hiera should be setup to send logs to the right places.

Can you at least add an example in the `README.md` of how to forward logs an example log server? I think that would be a common use for this profile, and it is not obvious to me at this point of how to enable that via this profile.

File: data/common.yaml [hunk 3]
```diff
@@ -46,6 +46,61 @@ profile_monitoring::telegraf::outputs:
...
+            port: "1514"
+            queue.FileName: "syslog-security-buffer"
+            queue.SaveOnShutdown: "on"
+            queue.Type: "LinkedList"
+            queue.size: "1000000"
+            queue.maxdiskspace: "10000000000"
+            queue.syncqueuefiles: "on"
+            Action.ResumeInterval: "30"
+            Action.ResumeRetryCount: "-1"
+            timeout: "6"
```
Comment (billglick) [inline data/common.yaml:102]: I think we should forward logs to the NCSA Security loghost by default. My thinking is that anything that gets setup should forward to security by default. If a project is setting up their own control repo from the template, they can then choose to change the forwarding to their own relay host if they want, but by default we start off forwarding them centrally. The counter argument is that it advertises NCSA’s primary loghost hostname and listening port.

**Comment by billglick** [comment]
The latest changes include:
- Update to `ncsa/profile_rsyslog` v0.1.1 (updated README with example)
- Inclusion of default data to `config_rulesets` to forward to NCSA's default security loghost
- Update to `puppet/rsyslog` to version 5.2.0

File: data/common.yaml (hunk 3 above)
Comment (andylytical) [reply data/common.yaml:102]: Can we take this opportunity to move this "default" data to the profile?

**Comment by billglick** [comment]
The latest changes include:
- Add new rsyslog profile to Puppetfile and added client profile to base.pp
- Update puppet/rsyslog to version 5.2.0
- Move default profile_rsyslog data to ncsa/profile_rsyslog module
- Update ncsa/profile_rsyslog to tag v0.1.2

**Review by andylytical — APPROVED** [review]

**Pull request merged**